CMakeLists.txt

```diff
@@ -29,6 +29,8 @@ endif()
 #add_compile_options(-fsanitize=address)
 #add_link_options(-fsanitize=address)
 
+add_subdirectory(gfx)
```

Comment:
why not create a toplevel cmake config?

Reply:
didn't want to clog up the main CMakeLists and also I was compiling the code within this library with fno-exceptions so didn't want that to leak into other libraries

Reply:
ok